scripts/azureml-assets/azureml/assets/model/model_utils.py

```diff
@@ -151,3 +153,52 @@ def prepare_model(spec_path, model_config, registry_name, temp_dir):
     except Exception as e:
         logger.log_error(f"prepare model failed for {spec_path}. Error {e}")
         return None, False
+
+
+def update_model_metadata(
+        model_name: str, 
+        model_version: str, 
+        spec_path: Path, 
+        model_config: assets.ModelConfig,
+        registry_name: str, 
+        update: AssetVersionUpdate = None,
+    ):
+    """Update the mutable metadata of already registered Model asset."""
+    try:
+        ml_client = MLClient(credential= AzureCliCredential(), registry_name=registry_name)
```

Comment:
we can get rid of mlclient operation here and just make use of model_config and spec_path

Reply:
az ml model update does not use spec path as such ref- https://learn.microsoft.com/en-us/cli/azure/ml/model?view=azure-cli-latest#:~:text=Global%20Parameters-,az%20ml%20model%20update,-Update%20a%20model


Also if we use CLI, It uses commands like 

az ml model update -- set.tags.tag1=value1 -- set.tags.tag2=value1 --remove.tags.tag3

for multiple tags it would be very cumbersome to maintain, similarly for removing some tags as well. It will be more error prone.It will be very long command.

Using sdk makes it quite simple just update the tags for model object and just update.

Reply:
yeah make sense. Thanks Hrishikesh

Reply:
we would need mlclient to fetch current models details and though we can do that via cli as well, but unnecessary parsing would be needed,. For tags, we would need to build command str iterating through all the keys. Current approach make sense to me. The reason I asked to check, is because we had faced some issue in creating mlclient earlier (that could be because if using DefaultAzureCredential. I am not very sure of its reason as of now). But since we are also creating AzureCLICredential for registry utils and its working, I guess we should be okay